models/clm/shell_scripts/cesm2_2/assimilate.csh

```diff
@@ -0,0 +1,401 @@
+# and are used to determine the domain metadata and *shape* of the variables.
+# "clm_vector_history_filename" is used to determine the shape of the
+# variables required to be read from the vector history file. If there are no
+# vector-based history variables, 'clm_vector_history_filename' is not used.
+#
+# &filter_nml
+#     async                   = 0,
+#     obs_sequence_in_name    = 'obs_seq.out'
+#     obs_sequence_out_name   = 'obs_seq.final'
+#     init_time_days          = -1,
+#     init_time_seconds       = -1,
+#     first_obs_days          = -1,
+#     first_obs_seconds       = -1,
+#     last_obs_days           = -1,
+#     last_obs_seconds        = -1,
+#     input_state_file_list   = "restart_files.txt",
+#                               "history_files.txt",
+#                               "vector_files.txt"
+#     output_state_file_list  = "restart_files.txt",
+#                               "history_files.txt",
+#                               "vector_files.txt"
+# &model_nml
+#     clm_restart_filename        = 'clm_restart.nc'
+#     clm_history_filename        = 'clm_history.nc'
+#     clm_vector_history_filename = 'clm_vector_history.nc'
+# &ensemble_manager_nml
+#     single_restart_file_in  = .false.
+#     single_restart_file_out = .false.
+#=========================================================================
+# clm always needs a clm_restart.nc, clm_history.nc for geometry information, etc.
+# it may or may not need a vector-format history file - depends on user input
+
+${REMOVE} restart_files.txt history_files.txt vector_files.txt
+
+foreach FILE ( ${CASE}.clm2_*.r.${LND_DATE_EXT}.nc )
+
+   # create unique output filename for the copy that has the indeterminate
+   # values replaced by the _FillValue. The copies are the files that will
+   # be used to construct the DART vector.
+   set OUTPUT = `echo $FILE | sed -e "s/${CASE}.//"`
+
+   ${COPY} $FILE clm.nc
+   ${EXEROOT}/clm_to_dart >& /dev/null
+   ${MOVE} clm.nc $OUTPUT
+
+end
+
+ls -1         clm2_*.r.${LND_DATE_EXT}.nc  >! restart_files.txt
+ls -1 ${CASE}.clm2_*.h0.${LND_DATE_EXT}.nc >! history_files.txt
+ls -1 ${CASE}.clm2_*.h2.${LND_DATE_EXT}.nc >! vector_files.txt
+
+#=========================================================================
+# Block 6: Actually run the assimilation.
+#=========================================================================
+
+echo "`date` -- BEGIN FILTER"
+${MPI_RUN_COMMAND} ${EXEROOT}/filter || exit 6
+echo "`date` -- END FILTER"
+
+#=========================================================================
+# Block 7: Put the DART posterior into the CLM restart file. The CLM
+# restart file is also the prior for the next forecast.
+#=========================================================================
+
+foreach RESTART ( ${CASE}.clm2_*.r.${LND_DATE_EXT}.nc )
+
+   set POSTERIOR = `echo $RESTART | sed -e "s/${CASE}.//"`
+
+   ${LINK} $POSTERIOR dart_posterior.nc
+   ${LINK} $RESTART   clm_prior.nc
```

Comment:
The posterior updates will not be passed onto the restart files unless
`${LINK} $RESTART clm_prior.nc` is changed to
`${LINK} $RESTART clm_restart.nc`